sys/stdio/stdio.c

```diff
@@ -61,6 +68,21 @@ void stdio_close(void) {
 #define MAYBE_WEAK __attribute__((weak))
 #endif
 
+int stdio_write_all(const void* buffer, size_t len)
+{
+    while (len > 0) {
+        ssize_t written = stdio_write(buffer, len);
+        if (written < 0) {
+            return written;
+            break;
+        }
+        buffer += written;
+        len -= written;
+    }
```

Comment:
You are already doing this in `stdio_write()`

Reply:
Only when using the dispatch backend, because it can't practically return incomplete writes. Other backends may still return incomplete writes, and all the callers that were switched to stdio_write_all in this PR previously didn't do this loop manually.

Reply:
Then what do you still need `stdio_write()` for?

Reply:
Because this allows applications to decide to slow things down if prints get chunked up too much, or decide to print less.

There is the alternative to make our stdio_write always behave like stdio_write_all, and block all the way through. That would be simpler, just less POSIXy, and I've clung (past tense) to the idea that at some point there'd have been a good reason to do things that way.

Reply:
No application will ever call stdio_write(), this hooks into libc.